### PR TITLE
Adjust LiveChat API Service offering

### PR DESCRIPTION
**Pull request by equalsgibson**

### CODE REVIEW

**Comment by equalsgibson** [comment]
Hey @aaronellington, I've refactored some small bits here and there. I am hoping to continue this, and also propose adding gobwas (ws) for the LiveChat Streaming section

Alternatively, we could try and roll our own WS lib using the std lib - I don't mind either way.

Let me know your thoughts on the above

File: zendesk/types.go [hunk 1]
```diff
@@ -96,6 +130,7 @@ type (
 	BrandID                  uint64
 	CategoryID               uint64
 	ChatAccountID            uint64
+	ChatDepartmentID         uint64
```
Comment (aaronellington) [inline zendesk/types.go:133]: I think we can just got rid of this and use GroupID

File: zendesk/service.go [hunk 2]
```diff
@@ -119,17 +119,27 @@ func NewService(
...
+				},
+				departmentService: &DepartmentService{
+					client: c,
+				},
+			},
+			realTimeChatService: &RealTimeChatService{
+				restService: &RESTService{
+					client: c,
+				},
+				websocketStreamingService: &WebsocketStreamingService{
```
Comment (aaronellington) [inline zendesk/service.go:138]: Can we rename all these from WebsocketStreamingService to RealTimeChatStreamingService so the exported type is associated with the the chat side of the house and not the ticket/support side of things.

File: zendesk/service.go [hunk 3]
```diff
@@ -119,17 +119,27 @@ func NewService(
...
+					client:           c,
+					agentStatesMutex: &sync.Mutex{},
+					agentStates:      AgentStates{},
+				},
+				departmentService: &DepartmentService{
+					client: c,
+				},
+			},
+			realTimeChatService: &RealTimeChatService{
+				restService: &RESTService{
```
Comment (aaronellington) [inline zendesk/service.go:135]: Same with this. Let's call it RealTimeChatRestService.

**Review by aaronellington — CHANGES_REQUESTED** [review]

**Comment by aaronellington** [comment]
Thanks! Looks pretty good! There is a lot here. I've added a few comments but we will chat about this tomorrow morning. 👍

File: zendesk/types.go (hunk 1 above)
Comment (equalsgibson) [reply zendesk/types.go:133]: Agreed, changing now - Originally thought that these were distinct (maybe they were at one point?) but they appear to be the same

File: zendesk/client.go [hunk 4]
```diff
@@ -14,6 +14,11 @@ import (
 	"time"
 )
 
+const (
```
Comment (equalsgibson) [inline zendesk/client.go:17]: These changes for the consts were required by the linter

**Comment by equalsgibson** [comment]
Hi @aaronellington, I've added some additional testing. For now, I am planning on leaving the WSS as a stub function (just the service), and will complete the work for that in a separate PR (as it touches a different area of the api). 

I believe that I have made the requested changes to the PR - let me know your thoughts and if I am good to merge

**Review by aaronellington — APPROVED** [review]

**Pull request merged**